src/components/AppSettingsDialog/AppSettingsDialog.vue

```diff
@@ -0,0 +1,308 @@
+ - @copyright Copyright (c) 2020 Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @author Marco Ambrosini <marcoambrosini@pm.me>
+ -
+ - @license GNU AGPL version 3 or any later version
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+<docs>
+Just nest the `AppSettingSections` component into `AppSettingsDialog`,
+providing the section's title prop. You can put your settings within each
+`AppSettingsSection` component.
+
+```vue
+<template>
+	<div>
+		<button @click="settingsOpen = true">Show Settings</button>
+		<AppSettingsDialog :open.sync="settingsOpen" >
+			<AppSettingsSection title="Example title 1">
+				Some example content
+			</AppSettingsSection>
+			<AppSettingsSection title="Example title 2">
+				Some more content
+			</AppSettingsSection>
+		</AppSettingsDialog>
+	</div>
+</template>
+
+<script>
+export default {
+	data() {
+		return {
+			settingsOpen: false,
+		}
+	},
+}
+</script>
+```
+</docs>
+
+<script>
+import Modal from '../Modal'
+import isMobile from '../../mixins/isMobile'
+import debounce from 'debounce'
+
+export default {
+
+	name: 'AppSettingsDialog',
+
+	components: {
+		Modal,
+	},
+
+	mixins: [isMobile],
+
+	props: {
+		/**
+		 * Determines the open / closed state of the modal
+		 */
+		open: {
+			type: Boolean,
+			required: true,
+		},
+		/**
+		 * Hides the navigation on desktop if true
+		 */
+		hideNavigation: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			selectedSection: '',
+			linkClicked: false,
+			addedScrollListener: false,
+			scroller: null,
+		}
+	},
+
+	computed: {
+
+		hasNavigation() {
+			if (this.isMobile || this.hideNavigation) {
+				return false
+			} else {
+				return true
+			}
+		},
+
+		settingsNavigationAriaLabel() {
+			return t('Settings navigation')
+		},
+	},
+
+	mounted() {
+		// Select first settings section
+		this.selectedSection = this.$slots.default[0].componentOptions.propsData.title
+	},
+
+	updated() {
+		if (!this.scroller) {
+			// Get the scroller element
+			this.scroller = this.$el.querySelector('.app-settings__content')
+		}
+		if (this.scroller && !this.addedScrollListener) {
+			this.scroller.addEventListener('scroll', debounce(() => {
+				this.handleScroll()
+			}), 300)
+			this.addedScrollListener = true
+		}
+	},
+
+	beforeDestroy() {
+		document.removeEventListener('scroll', this.handleScroll)
```

Comment:
```suggestion
		this.scroller.removeEventListener('scroll', this.handleScroll)
```

Also, won't work as you're using another function on line 121. You're passing debounce and not `this.handleScroll`. The references are different

Reply:
Good catch, that was from before I added the debounce... So I passed in the whole function as it's defined above, good?

Reply:
No, you need to have the same reference, so put the debounce inside methods :)

Reply:
ALso, please rebase and squash everything into one commit! :rocket: